src/basekernels/wiener.jl

```diff
@@ -75,4 +75,42 @@ function (::WienerKernel{3})(x, y)
         ( 5 * max(X, Y)^2 + 2 * X * Y + 3 * minXY^2 )
 end
 
+function kernelmatrix(::WienerKernel{I}, x::ColVecs, y::ColVecs) where I
+    validate_inputs(x, y)
+    X = sqrt.(sum(x.X .* x.X; dims=1))
+    Y = sqrt.(sum(y.X .* y.X; dims=1))
+    minXY = min.(permutedims(X), Y)
```

Comment:
Can't we just use
```suggestion
    X = sum(x.X .* x.X; dims=1)
    Y = sum(y.X .* y.X; dims=1)
    minXY = sqrt.(min.(permutedims(X), Y))
```
here, or maybe even not take the square root at all in cases where we compute (mostly) `minXY.^2` and `minXY.^4` such as `I == 3`?

Reply:
We need the squareroot-ed X and Y for I = 2 and 3. We also have odd powers of X and Y for I = 1, 2, 3.

Reply:
Sure but one could specialize on `I` (similar to the scalar definitions which could be changed as well). In any case, taking the square root only once (as done in the suggested change) is definitely more efficient than taking the square root of both `X` and `Y` separately.

Reply:
I am slightly confused. Taking square root for X and Y separate results in `2n` operations whereas taking squareroot on `minXY` would need `n*n` operations. Am I missing something?

Reply:
Oh haha, of course you're right! I completely missed the `permutedims` :sweat_smile: So the comment does not apply to the `kernelmatrix` implementation but rather to the existing one with individual samples.